### PR TITLE
fix warnings reported by ErrorProne (second PR)

### PR DESCRIPTION
**Pull request by zaki50**
@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java [hunk 1]
```diff
@@ -760,13 +760,16 @@ public void mutableMethodsOutsideTransactions() {
...
+                    default:
+                        // use default exception
                 }
             }
 
             try {
                 switch (method) {
                     case DELETE_ALL: collection.deleteAllFromRealm(); break;
-                    case ADD_OBJECT: collection.add(new AllJavaTypes());
+                    case ADD_OBJECT: collection.add(new AllJavaTypes()); break;
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java]: Would you be willing to put these `break`s on new lines?  It would be more consistent, if less compact.

File: realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java [hunk 2]
```diff
@@ -760,13 +760,16 @@ public void mutableMethodsOutsideTransactions() {
                     case REMOVE_ALL:
                     case RETAIN_ALL:
                         expected = UnsupportedOperationException.class;
+                        break;
+                    default:
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java]: I'm not sure I see the advantage in adding an explicit default clause...

File: realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java [hunk 3]
```diff
@@ -847,15 +850,18 @@ public Boolean call() throws Exception {
                         case REMOVE_ALL:
                         case RETAIN_ALL:
                             expected = UnsupportedOperationException.class;
+                            break;
+                        default:
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java]: Again, I'm not sure why the `default`

File: realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java [hunk 4]
```diff
@@ -564,7 +564,8 @@ public void looperThreadQuitsLooperEarly() throws InterruptedException {
 
         // Starts background looper and let it hang.
         ExecutorService executorService = Executors.newSingleThreadExecutor();
-        executorService.submit(new Runnable() {
+        //noinspection unused
+        final Future<?> future = executorService.submit(new Runnable() {
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java]: Why keep the `future`?  It is never used, I believe...

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java [hunk 5]
```diff
@@ -99,14 +99,14 @@ private void populateTestRealm(Realm testRealm, int dataSize) {
             allTypes.setColumnBinary(new byte[]{1, 2, 3});
             allTypes.setColumnDate(new Date(DECADE_MILLIS * (i - (dataSize / 2))));
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java]: If you would be willing, I would prefer that all of those lower case 'f' float indicators were uppercased.  It isn't so important with the 'f', but getting in the habit of using upper case prevents having to distinguish between `54321` and `5432l`

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 6]
```diff
@@ -108,7 +109,8 @@ public Thread newThread(Runnable runnable) {
                             return new Thread(runnable, threadName);
                         }
                     });
-                    executorService.submit(new Runnable() {
+                    //noinspection unused
+                    final Future<?> submit = executorService.submit(new Runnable() {
```
Comment (bmeike) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: Yea... I don't get this.  Why not just ignore it?

File: realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java (hunk 2 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java]: The `case`s in this `switch` are not covering all values in the `Enum` and that causes some warnings of ErrorProne.
This `default` means all other `case`s can be ignored.

File: realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java (hunk 1 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java]: I aligned it to existing code.
I think reformatting should be done in separate PR.

File: realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java (hunk 4 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java]: `ErrorProne` reports a warning that the return value of `submit()` should not be ignored.
To suppress that, just assigning the return value to unused local variable.

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java (hunk 5 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java]: I agree with that, but this PR is focusing on fixing warnings from `ErrorProne`.
We should have another PR to fix  that.

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java (hunk 6 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: same

File: realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java (hunk 4 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java]: Of course we can suppress the ErrorProne warning directly, but it affects entire method.
I think suppressing scope should be as narrow as possible.

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java (hunk 5 above)
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java]: That is actually a great argument. I think right now we are fairly inconsistent, but I'm all for enforcing uppercasing the type suffix 👍
